app/services/asset_manager/attachment_updater.rb

```diff
@@ -0,0 +1,38 @@
+class AssetManager::AttachmentUpdater
+  def self.call(
+    attachment_data,
+    access_limited: false,
+    draft_status: false,
+    link_header: false,
+    redirect_url: false,
+    replacement_id: false
+  )
+    updates = []
+
+    updates += AccessLimitedUpdates.call(attachment_data).to_a if access_limited
+    updates += DraftStatusUpdates.call(attachment_data).to_a if draft_status
+    updates += LinkHeaderUpdates.call(attachment_data).to_a if link_header
+    updates += RedirectUrlUpdates.call(attachment_data).to_a if redirect_url
+    updates += ReplacementIdUpdates.call(attachment_data).to_a if replacement_id
+
+    self.combined_updates(updates).each(&:call)
+  end
+
+  def self.combined_updates(updates)
+    grouped_updates = updates.group_by do |update|
+      [update.attachment_data, update.legacy_url_path]
+    end
+
+    grouped_updates.map do |attachment_data_and_legacy_url_path, updates_to_combine|
```

Comment:
A slightly neater thing here, you can deconstruct this param:
```
grouped_updates.map do |(attachment_data, legacy_url_path), updates_to_combine|
```

Reply:
Good idea, have amended!